4-star-nosed_mole/multipart_upload_specifications.md

```diff
@@ -0,0 +1,108 @@
+# Up-/Download Client & Production POC (Star-Nosed Mole) - Multipart Uploads
+
+This is an addition to the Star-Nosed mole epic to adress challenges in multipart uploads
+
+## Scope:
+A scope definition can be found here: https://wiki.verbis.dkfz.de/x/fAFzCQ
+
+## User Journeys
+
+The addition to this epic covers the following user journeys:
+
+### Multipart Data Upload:
+![Data Upload](./images/multipart_data_upload.png)
+
+A Data Submitter specifies the file ID, the URL to the Upload Contoller API, and the file path on the local file system using the CLI interface of the client (1.0). Internally, the CLI client translates the user-defined data into a request to the Upload Contoller API to obtain an S3 upload id (1.1). The client sends a request to the Upload Controller for each individual part of the uploaded file and recieves a presigned post for each individual part (1.2). The client reads multiple parts from the source file and uploads them as a stream using the pre-signed post (1.3). Once the upload has been completed the client sends a confirmation to the Upload Controller API (1.4).
```

Comment:
```suggestion
A Data Submitter specifies the file ID, the URL to the Upload Contoller API, and the file path on the local file system using the CLI interface of the client (1.0). Internally, the CLI client translates the user-defined data into a request to the Upload Contoller API to obtain an upload id (1.1). The client sends a request to the Upload Controller for each individual part of the uploaded file and recieves a presigned post for each individual part (1.2). The client reads multiple parts from the source file and uploads them as a stream using the pre-signed post (1.3). Once the upload has been completed the client sends a confirmation to the Upload Controller API thereby specifying the eTag for each file part (1.4).
```

We can just stay generic and say there is an upload ID, we might want to only keep the actual ID we get from S3 private in our database and expose another ID to the public.